### PR TITLE
BUG: Fix np.average for object arrays

### PR DESCRIPTION
**Pull request by warut-vijit**
Fixes type incompatibility with Decimal arrays (#8696), idea for fix by user @NateyB

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -1135,7 +1135,7 @@ def average(a, axis=None, weights=None, returned=False):
             wgt = wgt.swapaxes(-1, axis)
 
         scl = wgt.sum(axis=axis, dtype=result_dtype)
```
Comment (eric-wieser) [inline numpy/lib/function_base.py]: I think `keepdims=True` would be enough to fix this

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -1135,7 +1135,7 @@ def average(a, axis=None, weights=None, returned=False):
...
-        if (scl == 0.0).any():
+        if (type(scl)==np.bool_ and (scl == 0.0).any()) or scl == 0.0:
```
Comment (eric-wieser) [inline numpy/lib/function_base.py]: This should probably just be `np.any(scl == 0.0)` to make it work

**Comment by eric-wieser** [comment]
Commit message should start with `BUG:`

**Comment by eric-wieser** [comment]
You should add a test using the `Decimal` example here too. And squash your commits, starting the resulting one with the message `BUG: `

**Comment by warut-vijit** [comment]
Cleaned up with requested changes in pull request #8741

**Comment by eric-wieser** [comment]
There's no need to create a new PR like that - you can just force push to replace the old one

**Pull request closed without merging**